homeassistant/components/history_stats/sensor.py

```diff
@@ -115,7 +149,12 @@ def __init__(
         self._end = end
         self._type = sensor_type
         self._name = name
-        self._unit_of_measurement = UNITS[sensor_type]
+        self._precision = precision
+        self._time_unit = time_unit
+        # use UNITS dict for all but 'time' sensors, which use the config variable time_unit
+        self._unit_of_measurement = (
+            UNITS[sensor_type] if sensor_type != CONF_TYPE_TIME else time_unit
+        )
```

Comment:
so UNITS[CONF_TYPE_TIME] is never used, right ?

Reply:
it is used to validate `CONF_TIME_UNIT` (see L80-82), which is provided by users via the new `time_unit:` configuration variable or defaults to `h` (hours).  please notice that the addition and usage of the `UNITS` dict was not my idea.  I simply adapted the implementation of the new configuration variables to the previous code.